### PR TITLE
Add new_cops rake task

### PR DESCRIPTION
**Pull request by tejasbubane**
To generate new cop

Like https://github.com/rubocop-hq/rubocop-rspec/blob/master/lib/rubocop/cop/rspec_cops.rb

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -36,3 +36,32 @@ task default: %i[
...
+  generator.inject_config(config_file_path: 'config/default.yml',
+                          version_added: bump_minor_version)
+
+  puts generator.todo
+end
+
+def bump_minor_version
+  major, minor, _patch = RuboCop::Minitest::VERSION.split('.')
+
+  "#{major}.#{minor.succ}.0"
```
Comment (koic) [inline Rakefile:66]: Patch version is not required.

```suggestion
  "#{major}.#{minor.succ}"
```

File: Rakefile [hunk 2]
```diff
@@ -36,3 +36,32 @@ task default: %i[
...
+  github_user = `git config github.user`.chop
+  github_user = 'your_id' if github_user.empty?
+
+  generator = RuboCop::Cop::Generator.new(cop_name, github_user)
+
+  generator.write_source
+  generator.write_spec
+  generator.inject_require(root_file_path: 'lib/rubocop/cop/minitest_cops.rb')
+  generator.inject_config(config_file_path: 'config/default.yml',
+                          version_added: bump_minor_version)
```
Comment (koic) [inline Rakefile:58]: RuboCop Minitest's max of `Layout/LineLength` is `120`.
https://github.com/rubocop-hq/rubocop-minitest/blob/v0.9.0/.rubocop.yml#L50-L51

One line will be enough.

```suggestion
  generator.inject_config(config_file_path: 'config/default.yml', version_added: bump_minor_version)
```

File: Rakefile [hunk 3]
```diff
@@ -36,3 +36,32 @@ task default: %i[
...
+    exit!
+  end
+
+  github_user = `git config github.user`.chop
+  github_user = 'your_id' if github_user.empty?
+
+  generator = RuboCop::Cop::Generator.new(cop_name, github_user)
+
+  generator.write_source
+  generator.write_spec
```
Comment (koic) [inline Rakefile:55]: RuboCop Minitest uses Minitest. Not RSpec. Probably requires a template for Minitest.

**Comment by duduribeiro** [comment]
this is a nice idea 💯

File: Rakefile (hunk 3 above)
Comment (tejasbubane) [reply Rakefile]: I added test generation methods to rubocop's `Generator` class.

File: lib/rubocop/cop/generator.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          def test_registers_offense_when_using_bad_method
+            assert_offense(<<~RUBY)
+              bad_method
+              ^^^^^^^^^^ Use `#good_method` instead of `#bad_method`.
+            RUBY
+
+            assert_correction(<<~RUBY)
+              good_method
+            RUBY
+          end
```
Comment (koic) [inline lib/rubocop/cop/generator.rb:25]: Can you add a test template for good case?

File: lib/rubocop/cop/generator.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Source and spec generator for new cops
```
Comment (koic) [inline lib/rubocop/cop/generator.rb:5]:
```suggestion
    # Source and test generator for new cops
```

File: lib/rubocop/cop/generator.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    #
+    # This generator will take a cop name and generate a source file
+    # and test file when given a valid qualified cop name.
+    class Generator
+      TEST_TEMPLATE = <<~SPEC
```
Comment (koic) [inline lib/rubocop/cop/generator.rb:10]:
```suggestion
      TEST_TEMPLATE = <<~TEST
```

**Comment by tejasbubane** [comment]
@koic Changes done.

**Comment by koic** [comment]
Thanks!

**Pull request merged**